clients/samples/25_gemm_scale_a_b_vector_ext/sample_hipblaslt_gemm_scale_a_b_vector_ext.cpp

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (C) 2022-2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <hip/hip_runtime.h>
+#include <hipblaslt/hipblaslt-ext.hpp>
+#include <iostream>
+
+#include "helper.h"
+
+void simpleGemmScaleABVecExt(hipblasLtHandle_t  handle,
+                          hipblasOperation_t trans_a,
+                          hipblasOperation_t trans_b,
+                          int64_t            m,
+                          int64_t            n,
+                          int64_t            k,
+                          int64_t            batch_count,
+                          float&             alpha,
+                          float&             beta,
+                          void*              d_a,
+                          void*              d_b,
+                          void*              d_c,
+                          void*              d_d,
+                          void*              d_workspace,
+                          int64_t            max_workspace_size,
+                          hipStream_t        stream);
+
+int main()
+{
+    // This is an example using hipblaslt extension API: ScaleA & ScaleB & ScaleC
+    Runner<hipblaslt_f8_fnuz, hipblaslt_f8_fnuz, hipblasLtHalf, float, float> runner(
+        128, 128, 128, 1, 1.f, 1.f, 32 * 128 * 128);
+
+    std::vector<float> scale_a(128, 0.5f); // Scale A vector setting
+    std::vector<float> scale_b(128, 2.0f); // Scale B vector setting
+
+    std::cout << "Running with Scale A = [0.5, ..., 0.5], Scale B = [2.0, ..., 2.0]" << std::endl;
```

Comment:
It is better to use different value for each scale A element since this example is scale A/B VECTOR but not scalar.